mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/AbstractAttachmentMapperTest.java

```diff
@@ -60,46 +65,113 @@ public void tearDown() throws MailboxException {
+        String actualContentAsString = IOUtils.toString(actual.getContent().getInputStream(), Charsets.UTF_8);
+        assertThat(actualContentAsString).isEqualTo(expectedContent);
     }
 
     @Test
     public void getShouldWork() throws IOException {
         // Given
-        String blobId = "blobId";
-        String expected = "content";
-
-        attachmentMapper.add(blobId, IOUtils.toInputStream(expected, Charsets.UTF_8));
+        String expectedContent = "content";
+        InMemoryAttachmentId blobId = InMemoryAttachmentId.of(1);
+        saveAsNewAttachment(blobId, expectedContent);
 
         // When
-        InputStream attachment = attachmentMapper.get(blobId);
+        Attachment actual = attachmentMapper.get(blobId);
+        String actualContentAsString = IOUtils.toString(actual.getContent().getInputStream(), Charsets.UTF_8);
 
         // Then
-        assertThat(IOUtils.toString(attachment, Charsets.UTF_8)).isEqualTo(expected);
+        assertThat(actualContentAsString).isEqualTo(expectedContent);
     }
 
     @Test
     public void deleteShouldWork() throws IOException {
         // Given
-        String blobId = "blobId";
-        String expected = "content";
-
-        attachmentMapper.add(blobId, IOUtils.toInputStream(expected, Charsets.UTF_8));
+        String expectedContent = "content";
+        final InMemoryAttachmentId blobId = InMemoryAttachmentId.of(1);
+        saveAsNewAttachment(blobId, expectedContent);
 
         // When
-        attachmentMapper.remove(blobId);
+        attachmentMapper.delete(blobId);
 
         // Then
-        assertThat(attachmentMapper.get(blobId)).isNull();
+        try {
+            attachmentMapper.get(blobId);
+            fail("Should have thrown while accessing removed item !");
+        }
+        catch(IllegalArgumentException e) {
+        }
+    }
+
+    private void saveAsNewAttachment(AttachmentId blobId, String expectedContent) {
+        Content content = new ByteContent(expectedContent.getBytes(Charsets.UTF_8));
+        Attachment newAttachment = new SimpleAttachment(blobId, content);
+        attachmentMapper.put(newAttachment);
+    }
+
+    @Test
+    public void deleteShouldNotThrowWhenAttachmentIdNotFound() {
+        InMemoryAttachmentId blobId = InMemoryAttachmentId.of(-1);
+        attachmentMapper.delete(blobId);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void getShouldThrowWhenAttachmentIdNotFound() {
+        InMemoryAttachmentId blobId = InMemoryAttachmentId.of(-1);
+        attachmentMapper.get(blobId);
+    }
+
+    @Test(expected = NullArgumentException.class)
+    public void deleteShouldThrowWhenIdIsNull() {
+        attachmentMapper.delete(null);
     }
+
+    @Test(expected = NullArgumentException.class)
+    public void getShouldThrowWhenIdIsNull() {
+        attachmentMapper.get(null);
+    }
+
+    public static class InMemoryAttachmentId implements AttachmentId {
```

Comment:
Originally all implementations of attachment-related logic (mapper, ...) were templated on interface AttachmentId, just like MailboxManager is templated on the MailboxId.
This specific (and unique) implementation is for testing purpose.
I guess sharing same class will maybe emerge once we have many store implementations dealing with similar IDs, but we're not sure about that yet.